src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependencyNodeId.cs

```diff
@@ -96,9 +108,12 @@ public override string ToString()
             builder.Append(";");
             builder.Append(UniqueToken);
 
+            builder.Append(";");
+            builder.Append(ContextProject);
+
             builder.Append("]");
 
-            return builder.ToString();
+            return HttpUtility.UrlEncode(builder.ToString());
```

Comment:
Can you not pull in System.Web, and instead use https://msdn.microsoft.com/en-us/library/system.net.webutility(v=vs.110).aspx.